extensions/amp-sidebar/1.0/component.js

```diff
@@ -190,45 +191,46 @@ export function SidebarToolbar({
   children,
   ...rest
 }) {
-  const ref = useRef();
-  const [matches, setMatches] = useState(false);
-  const [targetEl, setTargetEl] = useState(null);
+  const ref = useRef(null);
+  const [target, setTarget] = useState();
 
   useEffect(() => {
-    const window = ref.current?.ownerDocument?.defaultView;
-    if (!window) {
-      return;
-    }
-
-    const mediaQueryList = window.matchMedia(mediaQueryProp);
-    const updateMatches = () => setMatches(mediaQueryList.matches);
-    mediaQueryList.addEventListener('change', updateMatches);
-    setMatches(mediaQueryList.matches);
-    return () => mediaQueryList.removeEventListener('change', updateMatches);
-  }, [mediaQueryProp]);
+    const selector = `#${escapeCssSelectorIdent(toolbarTarget)}`;
+    const newTarget = document.querySelector(selector);
```

Comment:
We should retrieve `ownerDocument` from an existing ref, i.e. `ref.current?.ownerDocument` instead of access `document` directly.

Reply:
Thanks Caroline updated!